### ISSUE TITLE
Default emissivity disagreement

### ISSUE THREAD

**Issue by chadagreene** [alex-s-gardner/GEMB#16]
The [Gardner et al., 2023 manuscript](https://gmd.copernicus.org/articles/16/2277/2023/) says: 

> The snow–ice is assumed to be a grey body **(default emissivity = 0.97)** with the incoming longwave radiation absorbed within the top layer and outgoing longwave radiation calculated using the Stefan–Boltzmann constant ( W m−2 K−4).

However, `thermo.m` [currently](https://github.com/alex-s-gardner/GEMB/blob/518c7eac8915ec67fea360bb5b7b0666a57f7224/GEMB/thermo.m#L408) sets
```matlab
emissivity = 1.0; 
```
by default. Does the default emissivity need to be set to 0.97 in `termo.m`?

**Comment by NJSchlegel** [comment]
I think the default emissivity should be a black body.  Then the user will change teValue if they want a different assumption.   Note that when the emissivity is changed, the incoming longwave should take the emissivity into account, but it currently does not.  Need to update this.

**Comment by alex-s-gardner** [comment]
My preference if for default emissivity = 0.97 as we know that glacier emissivity is close to a black body but < 1.0. I can't remember where I got 0.97 from but I think typical values range between 0.95 and 0.99

**Comment by NJSchlegel** [comment]
The default emissivity was originally 1 in GEMB, hence the hardcode.  Now that we have an emissivity variable and input, it makes sense to think about using this lower value, which I got from some snow studies investigating what ice sheet emissivity is typically.  Some PREFIRE researchers also backed 0.97 as a good general choice.  In best practice, it probably should be based on things like the grain size, and possibly have a spectral value.  I will be working on testing these additional treatments in the coming months.  I think CESM uses 0.96.  If we change the value of the default, then I suggest we remove the hardcode from this function and force the user to provide a value, which would be something like 0.97 in the GEMB function itself for teValue (outside of the innards thermo.m code).

**Comment by alex-s-gardner** [comment]
+1 for what nicole suggests

**Comment by chadagreene** [comment]
I don't love the idea of forcing users manually set default values. The current implementation of the `GEMB` function requires users to manually enter more than 50 constants, variables, and preferences just to run a minimal working example. 

Most of the inputs can easily be set as defaults, and it'll be much more welcoming if it can run right out of the box with minimal inputs. The other part of that, though, is to make the default values easy to override, and perhaps even provide a few examples that _show_ how changing the defaults can impact the final results. An example or two will go a lot farther toward building confidence and intuition than putting the onus on users to decide on every default value that gets fed into the mystery box. 

It's important that it's easy for beginners to get started, and it's just as important to minimize clutter for frequent users. Because right now, the `MASTER_RUN.m` file, although very well organized and nicely explained, sort of resembles the _"It works, don't touch it"_ code I used in grad school and was always afraid I might accidentally adjust and not know how to set back to its default state.

**Comment by alex-s-gardner** [comment]
@chadagreene all variables should have a default value... this is easy to do in other languages but I never learned how to do it properly in Matlab. The user should only start changing defaults once they are comfortable and curious. I would consider the 50 values in the example the "default" values.

**Comment by NJSchlegel** [comment]
This sounds like you are discussing a bigger issue here, which maybe needs its own issue?  The issue is that someone running it should know what they are doing and at least get the physics.  That does not mean that it should not be more intuitive or overwhelming.   OK, but for the emissivity situation right now - its default should not be set in thermo.m.  Probably it should be set in the launching script or in GEMB itself (for example if teValue is not is passed in, we set it to 0.97).

**Comment by chadagreene** [comment]
Yep, I think we're all on the same page, and I'm working on a solution that elegantly address what we're all thinking. I will start an Issue in a few hours or tomorrow morning.

**Comment by chadagreene** [comment]
> all variables should have a default value... this is easy to do in other languages but I never learned how to do it properly in Matlab.

It's actually pretty elegant with [arguments](https://www.mathworks.com/help/matlab/ref/arguments.html) in Matlab. For example, 
```matlab
function myRectangle(X,Y,options)
    arguments
       X double
       Y double
       options.LineStyle (1,1) string = "-" 
       options.LineWidth (1,1) {mustBeNumeric} = 1
    end
    % Function code
    ...
end
```
makes any of these calls possible: 
```matlab 
myRectangle(4,5)
myRectangle(4,5,LineStyle=":",LineWidth=2)
myRectangle(4,5,LineWidth=2,LineStyle=":")
myRectangle(4,5,LineStyle=":")
myRectangle(4,5,LineWidth=2)
```
It's a pretty powerful syntax that allows any non-ambiguous abbreviation (lke `linew` for `LineWidth`) and can do optional data type checks, etc. I think it'll be good for letting GEMB users declare optional inputs easily and intuitively.
